packages/store/test/perf-patterns.js

```diff
@@ -0,0 +1,259 @@
+    yesPatterns: [
+      makeCopySet([4, 3]),
+      M.gte(makeCopySet([])),
+      M.lte(makeCopySet([3, 4, 5])),
+      M.set(),
+      M.setOf(M.number()),
+    ],
+  },
+  {
+    specimen: makeCopyBag([
+      ['a', 2n],
+      ['b', 3n],
+    ]),
+    yesPatterns: [
+      M.gt(makeCopyBag([])),
+      M.gt(makeCopyBag([['a', 2n]])),
+      M.gt(
+        makeCopyBag([
+          ['a', 1n],
+          ['b', 3n],
+        ]),
+      ),
+      M.bag(),
+      M.bagOf(M.string()),
+      M.bagOf(M.string(), M.lt(5n)),
+    ],
+  },
+  {
+    specimen: makeCopyMap([
+      [{}, 'a'],
+      [{ foo: 3 }, 'b'],
+    ]),
+    yesPatterns: [
+      // M.gt(makeCopyMap([])), Map comparison not yet implemented
+      M.map(),
+      M.mapOf(M.record(), M.string()),
+    ],
+  },
+  {
+    specimen: makeTagged('mysteryTag', 88),
+    yesPatterns: [M.any(), M.not(M.pattern())],
+  },
+  {
+    specimen: makeTagged('match:any', undefined),
+    yesPatterns: [M.any(), M.pattern()],
+  },
+  {
+    specimen: makeTagged('match:any', 88),
+    yesPatterns: [M.any(), M.not(M.pattern())],
+    noPatterns: [[M.pattern(), 'match:any payload: 88 - Must be undefined']],
+  },
+  {
+    specimen: makeTagged('match:remotable', 88),
+    yesPatterns: [M.any(), M.not(M.pattern())],
+  },
+  {
+    specimen: makeTagged('match:remotable', harden({ label: 88 })),
+    yesPatterns: [M.any(), M.not(M.pattern())],
+  },
+  {
+    specimen: makeTagged('match:recordOf', harden([M.string(), M.nat()])),
+    yesPatterns: [M.pattern()],
+  },
+  {
+    specimen: makeTagged(
+      'match:recordOf',
+      harden([M.string(), Promise.resolve(null)]),
+    ),
+    yesPatterns: [M.any(), M.not(M.pattern())],
+  },
+]);
+
+const makeBatch = testCases => {
+  const specimens = [];
+  const patterns = [];
+  for (const { specimen, yesPatterns } of testCases) {
+    for (const yesPattern of yesPatterns) {
+      specimens.push(specimen);
+      patterns.push(yesPattern);
+    }
+  }
+  return harden({ specimens, patterns });
+};
+
+const repetitions = 200;
+
+// export NODE_OPTIONS="--jitless --cpu-prof --heap-prof --expose-gc"
+// NODE_OPTIONS="--jitless" node --cpu-prof --prof test/perf-patterns.js
+// node --prof-process isolate-0x140008000-36726-v8.log > processed.txt
+// node --prof test/perf-patterns.js
+// node --prof-process isolate-0x130008000-37133-v8.log > processed.txt
+
+const measure = async (title, specimen, yesPattern, factor = 1) => {
+  const times = factor * repetitions;
+  await gcAndFinalize();
+  const startAt = Date.now();
+  for (let j = 0; j < times; j += 1) {
+    // fit(specimen, yesPattern);
+    assert(matches(specimen, yesPattern), title);
+  }
+  const totalTime = Date.now() - startAt;
+  const perStep = totalTime / times;
+  measurements.push([title, perStep, Math.round(1000 / perStep)]);
+  console.log(
+    `test '${title}': ${times} repetitions took ${totalTime}ms;  each: ${perStep}ms`,
+  );
+};
+
+const issuerKit = makeIssuerKit('fungible');
```

Comment:
using `Far` is sufficient to make things that match the various issuer patterns for measurement purposes. So you don't need this dependency

Reply:
Done